_rules/links-identical-name-equivalent-purpose-b20e66.md

```diff
@@ -332,7 +332,7 @@ Both links resolve to [same resource][] after redirect, but the redirect is not
 
 #### Inapplicable Example 1
 
-These `a` and `area` elements have no `href` attribute.
+These `a` and `area` elements have no `href` attribute. These are not valid links.
```

Comment:
```suggestion
These `a` and `area` elements have no `href` attribute. Thus, they do not have a role of `link`.
```

Reply:
I'm not sure on this as the 'a' and 'area' are the role of link, but the 'href' is where it goes.

Reply:
According to the mappings: https://www.w3.org/TR/html-aam-1.0/#html-element-role-mappings (direct links to the elements don't work well… you need to scroll… 😞 ), the `a` and `area` elements have no role when there is no `href` attribute.
So, in that case, there is no element with a role of `link` (or inheriting from `link`), hence the rule is not applicable. I say, "role of `link`" is the technical term for "valid link" 😀

Reply:
"not work as expected" is a fairly strong statement, given that there are valid cases for `a` without `href`: https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-a-element (see the second paragraph after the green box + the first example). So, it might be a deliberate choice of the author to omit the `href` attribute, and the element would work exactly as expected.

(+ see the example at https://html.spec.whatwg.org/multipage/image-maps.html#image-maps for an `area` without `href` where the lack of `href` is exactly what is intended, namely to make the "hole" in the red square not clickable)

Reply:
So in plain text how would you define a link without a role?

Reply:
Well, that's the point. It's not a link.
It's an `a`/`area` element without an `href` attribute and therefore not a link (role has little to do here, it's not a link for ARIA (no role), but it's not a link for HTML in the first place (not clickable, …)

Maybe "Thus they are not links and do not have a role of `link`"?

Reply:
Yes - that would work 👍

Reply:
@Jym77, can I update the last suggestion to the change to commit? Or is it something for you to change?

Reply:
Please do it.
I normally prefer to let one person handle all changes on a PR, so that there is clear responsibility (not in term of "who to blame?", but rather in avoiding the "who should do that?" and streamlining the process).